### DISCUSSION TITLE
M.2 Coral Edge TPU in stock

### DISCUSSION THREAD

**Discussion by richardkuw** [blakeblackshear/frigate#4110] [General]
Just a heads up that Arrow has 83 of these in stock. This is the A+E slot part, so it would fit in an M.2 slot meant for a Wi-fi card, not a storage slot. Or use a PCIe Wi-fi card adapter to plug into a full size PCIe slot.

I have no connection with Arrow.
https://www.arrow.com/en/products/g650-04527-01/google-corporation

**Comment by git-coder988** [comment]
Thank you for this tip!  I know that this reply is a bit late...and Arrow
is now OOS...  But if you happen to find one of these (I actually have a
possible spare that I may sell on ebay), you can get it to work in a normal
PCIE slot.  I'm running Frigate in a Dell Optiplex 7040 (i5-6500) with full
size PCIE slots.

Here's what you need:
1. $25 Coral TPU with the A+E key
2. $8.59 M.2 2230 Key A A+E to mini PCIE WiFi Bluetooth Adapter for NGFF
AX200 AX210 Card - I got this one on eBay:
https://www.ebay.com/itm/283744526172
3. $12 PCI Express to Mini PCI Express Card Adapter Wireless Network
Ethernet LAN Mini PCI-E Riser PC to Laptop Network Card Adapter Converter
Antenna WiFi - Amazon:
https://www.amazon.com/gp/product/B07N2X62LQ/ref=ppx_yo_dt_b_asin_title_o02_s00?ie=UTF8&psc=1

NOTE:  There are a couple of comments in the Amazon listing stating that
the adapter does NOT work with Coral.  I can confirm that this
configuration DOES in fact work.  Just make sure that you install the
drivers as documented here:
https://coral.ai/docs/m2/get-started/#2a-on-linux

$ lspci -nn | grep Coral
03:00.0 System peripheral [0880]: Global Unichip Corp. Coral Edge TPU
[1ac1:089a]

My inference speed went from >100 to under 10 and the system CPU is
considerably lower.
